### PR TITLE
Added singularity to getNetlink and getFRR status information

### PR DESCRIPTION
**Pull request by Cellebyte**
* Concurrent requests for frr and netlink should only do a single backend call instead of multiple calls

### CODE REVIEW

File: pkg/monitoring/nl.go [hunk 1]
```diff
@@ -34,42 +37,77 @@ func NewNetlinkCollector() (Collector, error) {
...
+	collector.name = nlCollectorName
+	collector.logger = ctrl.Log.WithName("netlink.collector")
 
 	return &collector, nil
 }
 
-func (c *netlinkCollector) updateRoutes(ch chan<- prometheus.Metric) {
+func (c *netlinkCollector) getRoutes() []route.Information {
+	c.mu.Lock()
+	defer c.mu.Unlock()
```
Comment (schrej) [inline pkg/monitoring/nl.go:57]: locking shouldn't be required here

File: pkg/monitoring/nl.go [hunk 2]
```diff
@@ -34,42 +37,77 @@ func NewNetlinkCollector() (Collector, error) {
...
+func (c *netlinkCollector) updateRoutes(ch chan<- prometheus.Metric, routeSummaries []route.Information) {
+	for _, routeSummary := range routeSummaries {
+		ch <- c.routesFibDesc.mustNewConstMetric(float64(routeSummary.Fib), fmt.Sprint(routeSummary.TableID), routeSummary.VrfName, nl.GetProtocolName(routeSummary.RouteProtocol), routeSummary.AddressFamily)
 	}
 }
 
-func (c *netlinkCollector) updateNeighbors(ch chan<- prometheus.Metric) {
+func (c *netlinkCollector) getNeighbors() []nl.NeighborInformation {
+	c.mu.Lock()
+	defer c.mu.Unlock()
```
Comment (schrej) [inline pkg/monitoring/nl.go:73]: locking shouldn't be required here

**Review by schrej — CHANGES_REQUESTED** [review]

File: pkg/monitoring/collector.go [hunk 3]
```diff
@@ -94,11 +106,11 @@ func (DasSchiffNetworkOperatorCollector) Describe(ch chan<- *prometheus.Desc) {
 // Collect implements the prometheus.Collector interface.
 func (n DasSchiffNetworkOperatorCollector) Collect(ch chan<- prometheus.Metric) {
 	wg := sync.WaitGroup{}
-	wg.Add(len(n.Collectors))
 	for name, c := range n.Collectors {
+		wg.Add(1)
 		go func(name string, c Collector) {
 			execute(name, c, ch)
-			wg.Done()
+			defer wg.Done()
```
[the remark this answers is not in the document]
Comment (Cellebyte) [reply pkg/monitoring/collector.go:112]: fixed

File: pkg/monitoring/nl.go (hunk 1 above)
Comment (Cellebyte) [reply pkg/monitoring/nl.go]: fixed

**Review by schrej — APPROVED** [review]

**Pull request merged**